### ISSUE TITLE
MPI_CXX_DOUBLE_COMPLEX is an undeclared identifier

### ISSUE THREAD

**Issue by ArturSalamatin** [ddemidov/amgcl#236]
I am compiling the amgcl library as a .lib project in VS 2022 with C++17 standard (Release, x64).
It is a part of a larger solution with other project that would call the lib to solve the system of equations.
The compiler says the following

```
1>...HydroSolver\AMGSolver\amgcl\amgcl\mpi\util.hpp(102,40): error C2065: 'MPI_CXX_DOUBLE_COMPLEX': undeclared identifier
1>...HydroSolver\AMGSolver\amgcl\amgcl\mpi\util.hpp(107,40): error C2065: 'MPI_CXX_FLOAT_COMPLEX': undeclared identifier
```

How could I probably fix it?
I have already installed the MPI SDK from
`https://www.microsoft.com/en-us/download/confirmation.aspx?id=100593`
and did my best to follow this guide
`https://medium.com/geekculture/configuring-mpi-on-windows-10-and-executing-the-hello-world-program-in-visual-studio-code-2019-879776f6493f`

However, I do not expect to use the MPI API in the nearest future.
So, would you also be so kind to instruct on how to exclude the MPI dependencies from the amgcl library. (This is a priority, currently)

Thank you!

**Comment by ArturSalamatin** [comment]
[mpi.txt](https://github.com/ddemidov/amgcl/files/9329160/mpi.txt)

this is the mpi.h installed with the SDK

**Comment by ddemidov** [comment]
First, it looks like your version of MPI is too old. `MPI_CXX_DOUBLE_COMPLEX` and `MPI_CXX_FLOAT_COMPLEX` datatypes were introduced in MPI specification 2.2 and your mpi.h is versioned as 2.0. This should be fixed in the latest amgcl commit (3537d1f46f7279965526e05ded21830050661472).

Second, if you don't need the MPI functionality, then just don't include anything from `amgcl/mpi`.
